### ISSUE TITLE
iOS failing to match on Resource ID and Accessibility Text when there is another duplicate Accessibility Text

### ISSUE THREAD

**Issue by simon-gilmurray** [mobile-dev-inc/maestro#1404]
**Describe the bug**
iOS often fails to match on Resource ID and Accessibility Text when combined together - I believe this is happening if there is a duplicate Accessibility Text without an ID, although could be for another reason. Example:

```
- assertVisible:
    id: "viewAllSpacesBtn"
    text: "View all"
``` 

This fails although commenting either id or text out separately will both pass.

Hierarchy screenshot:

![Screenshot 2023-08-30 at 16 36 38](https://github.com/mobile-dev-inc/maestro/assets/74550141/1ca70119-9e00-4c94-955d-f2f3f305a9cb)

**Expected behavior**
Step should pass as a combination of the resource ID and text exist

**Environment information (please complete the following information):**

- Maestro version v1.31.0
- Platform: iOS 16
- Framework: Native
- Device model and OS version: iPhone 14
- Simulator, emulator or physical device: Simulator
- Host: Mac

**Comment by jg210** [comment]
I also see this on iOS (but not android), in particular the "commenting either id or text out separately will both pass" part. I still see this if get rid of all RN components apart from one Text, ruling out "duplicate Accessibility Text" as cause (as well as nesting Text components and too deep a hierarchy).

```
 Assert that (Optional) "testing 1" is visible... COMPLETED
Assert that (Optional) "testing 1", id: testing1 is visible... WARNED


 Warning: Assertion is false: "testing 1", id: testing1 is visible
Assert that (Optional) id: testing1 is visible... COMPLETED
```

Ran tests lots of times, so it's not just flaky.

More details in https://github.com/jg210/expo-experiments/pull/7#issuecomment-2732749198
